teamscale_precommit_client/precommit_client.py

```diff
@@ -104,11 +108,25 @@ def _upload_precommit_data(self):
         self.teamscale_client.branch = self.current_branch
 
         print("Uploading changes on branch '%s' in '%s'..." % (self.current_branch, self.repository_path))
-        precommit_data = PreCommitUploadData(uniformPathToContentMap=self.changed_files,
-                                             deletedUniformPaths=self.deleted_files)
+
+        changed_files_in_project = self._remove_changed_files_not_in_project_path(self.changed_files)
+        deleted_files_in_project = self._remove_deleted_files_not_in_project_path(self.deleted_files)
```

Comment:
Since the filter logic is specific to changed/deleted files, respectively, I would not pass the fields as arguments. You don't do this for the prefix either.

And then I would rename the methods to `_get_changed_files_in_project_path()` and `_get_deleted_files_in_project_path()` respectively.

Reply:
We'll need to pass the findings as arguments as soon as we have both `path-prefix` and `project-subpath`, though, since we are not working on the `changed_files` and `deleted_files` fields.

Reply:
`_filter_files_in_project_path(files)` then?